src/kubernetes.cc

```diff
@@ -1049,9 +1049,13 @@ void KubernetesReader::PodCallback(
     MetadataUpdater::UpdateCallback callback,
     const json::Object* pod, Timestamp collected_at, bool is_deleted) const
     throw(json::Exception) {
-  std::vector<MetadataUpdater::ResourceMetadata> result_vector =
-      GetPodAndContainerMetadata(pod, collected_at, is_deleted);
-  callback(std::move(result_vector));
+  try {
+    std::vector<MetadataUpdater::ResourceMetadata> result_vector =
+        GetPodAndContainerMetadata(pod, collected_at, is_deleted);
+    callback(std::move(result_vector));
+  } catch (const json::Exception& e) {
```

Comment:
I don't think this is the right fix for the situation -- this will simply ignore a JSON error. The assumption here is that if we got an invalid hunk of JSON, the rest of the hunks will also be garbage, so we shouldn't bother with the current connection -- it's better to re-establish the watch.